package/kedro_viz/server.py

```diff
@@ -70,35 +75,35 @@ def pretty_name(name):
     all_tags = set()
 
     for node in sorted(pipeline.nodes, key=lambda n: n.name):
-        task_id = "task/" + node.name.replace(" ", "")
+        task_id = _hash(str(node))
         nodes.append(
             {
                 "type": "task",
                 "id": task_id,
                 "name": getattr(node, "short_name", node.name),
-                "full_name": str(node),
+                "full_name": getattr(node, "_func_name", str(node)),
                 "tags": sorted(node.tags),
             }
         )
         all_tags.update(node.tags)
         for data_set in node.inputs:
             namespace = data_set.split("@")[0]
-            edges.append({"source": "data/" + namespace, "target": task_id})
+            edges.append({"source": _hash(namespace), "target": task_id})
             namespace_tags[namespace].update(node.tags)
         for data_set in node.outputs:
             namespace = data_set.split("@")[0]
-            edges.append({"source": task_id, "target": "data/" + namespace})
+            edges.append({"source": task_id, "target": _hash(namespace)})
             namespace_tags[namespace].update(node.tags)
 
     for namespace, tags in sorted(namespace_tags.items()):
+        is_param = bool("param" in namespace.lower())
```

Comment:
Shouldn't this be something like `bool("params:" in namespace.lower()) or "parameters" == namespace.lower()` (or we will display parameter icon whenever the dataset name included "param")?

Reply:
ideally the catalog should remember if it was added with add_feeddict and we should decide based on that
in the absence of that it's always been like this (it used to be done in the JS) and yes it could incorrectly pick up something that's not parameters